manifests/prepare/puppet_config.pp

```diff
@@ -26,7 +26,14 @@
       # Deprecated for global config
       'config_version', 'manifest', 'modulepath',
       # Settings that should be reset to defaults
-      'disable_warnings', 'vardir', 'rundir', 'libdir', 'confdir', 'ssldir', 'classfile'].each |$setting| {
+      'disable_warnings', 'vardir', 'rundir', 'libdir', 'confdir', 'ssldir', 'classfile']
+      # When upgrading to 1.4.x or later remove pluginsync
+      if versioncmp($puppet_agent::params::master_agent_version, '1.4.0') > 0 {
```

Comment:
Generally a good approach, but we should base this off the target version. That means adding a new parameter to this class to pass _package_version in from init.pp